proposed/sdk-analysis-level.md

```diff
@@ -0,0 +1,187 @@
+in his project file to unblock builds until he has time to investigate.
+
+---
+
+In both of these scenarios, the user is *not in control* of their environment. They can control their code,
+but they do not have the time to address the full set of problems introduced by an update. They use the new
+property to request older behavior, for a _limited time_, until they can address the issues. In addition,
+this single property was able to control the behavior of the SDK, the compilers, and any other tools that
+have been onboarded to the new scheme, without having to look up, comment out, or add many  `NoWarn` properties
+to their project files.
+
+
+## Requirements
+
+### Goals
+
+* Users have a unified way to manage diagnostics for all the tools in the SDK
+* Tooling has a way to understand what compatibility level a user explicitly wants
+* Users are able to upgrade across SDKs without being forced to immediately resolve warnings
+* Tools bundled in the SDK adopt `SdkAnalysisLevel` as a guideline
+
+<!--
+Provide a bullet point list of aspects that your feature has to satisfy. This
+includes functional and non-functional requirements. The goal is to define what
+your feature has to deliver to be considered correct.
+
+You should avoid splitting this into various product stages (like MVP, crawl,
+walk, run) because that usually indicates that your proposal tries to cover too
+much detail. Keep it high-level, but try to paint a picture of what done looks
+like. The design section can establish an execution order.
+-->
+
+### Non-Goals
+
+* Users keep using `SdkAnalysisLevel` indefinitely
+* Microsoft-authored tools delivered outside the SDK adhere to `SdkAnalysisLevel`
+* Non-Microsoft authored tools adhere to `SdkAnalysisLevel`
+
+<!--
+Provide a bullet point list of aspects that your feature does not need to do.
+The goal of this section is to cover problems that people might think you're
+trying to solve but deliberately would like to scope out. You'll likely add
+bullets to this section based on early feedback and reviews where requirements
+are brought that you need to scope out.
+-->
+
+## Stakeholders and Reviewers
+
+| Team | Representatives |
+| ---- | --------------- |
+| SDK | @marcpopmsft @dsplaisted |
+| MSBuild | @rainersigwald @ladipro |
+| NuGet | @aortiz-msft @nkolev92 |
+| Roslyn | @jaredpar @CyrusNajmabadi |
+| F# | @vzarytovskii @KevinRansom |
+| PM | @KathleenDollard @MadsTorgersen |
+
+## Design
+
+### Valid values of `SdkAnalysisLevel`
+
+The implementation of `SdkAnalysisLevel` is itself quite straightforward - the default value of `SdkAnalysisLevel` is always
+the stable 'SDK Feature Band' of the SDK that is currently being run, and the only valid values for the property are other SDK Feature Bands.
+An SDK Feature Band is a Semantic Version of the form `<Major>.<Minor>.<Patch>`, where the `<Patch>` version is a multiple of 100. Some
+valid feature bands for this discussion might be 6.0.100, 7.0.300, or 8.0.200.
```

Comment:
Is the scenario of preview / RC SDK updates excluded on purpose? It could be useful to say "keep me on SdkAnalysisLevel 9.0.100-preview.5" if Preview 6 introduced a change that broke my build. I understand that it would mean more internal bookkeeping but if we included the `SdkAnalysisLevel` overrides in anonymous telemetry, it could help guide product decisions within a single major.minor.patch release.

Reply:
Yes - preview-level changes aren't in scope for this document. Users opting into previews are inherently accepting changes, this proposal is mostly around how users of stable versions can mitigate behavioral changes temporarily.